torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -491,10 +491,12 @@ def inc_and_set(fut):
 # A custom Python class that contains a tensor, needed to see if we correctly
 # use the Python pickler to extract tensors from non-IValue-convertible types.
 class TensorWrapper:
-    __slots__ = ("tensor",)
+    __slots__ = ("tensor", "lock")
 
     def __init__(self, t):
         self.tensor = t
+        # Add one non-picklable field, to ensure it's ignored/skipped.
+        self.lock = Lock()
```

Comment:
Shall we add a test to ensure this works?

Reply:
Not sure I understand: this is the test. The method `test_cuda_future_can_extract_custom_class_with_cuda_tensor` will create an instance of this `TensorWrapper` class and pass it as a result to Future. By adding a lock as a field of TensorWrapper we would have caused a failure, but thanks to the fix this now works. Am I missing something?